src/GafferScene/ImageToPoints.cpp

```diff
@@ -0,0 +1,530 @@
+					}
+				}
+			);
+		}
+
+		pointsPrimitive->setNumPoints(
+			pointsPrimitive->variableData<V3fVectorData>( "P" )->readable().size()
+		);
+	}
+
+	return pointsPrimitive;
+}
+
+std::vector<ImageToPoints::ChannelMapping> ImageToPoints::channelMappings() const
+{
+	vector<ChannelMapping> result;
+
+	const string primitiveVariables = primitiveVariablesPlug()->getValue();
+	ConstStringVectorDataPtr positionChannelsData = positionPlug()->getValue();
+	const vector<string> &positionChannels = positionChannelsData->readable();
+	const string widthChannel = widthChannelPlug()->getValue();
+	const bool ignoreTransparent = ignoreTransparentPlug()->getValue();
+
+	unsigned int numPositionMappings = 0;
+	bool haveWidthMapping = false;
+	bool haveAlphaMapping = false;
+	ConstStringVectorDataPtr channelNamesData = imagePlug()->channelNamesPlug()->getValue();
+	for( const auto &channelName : channelNamesData->readable() )
+	{
+		ChannelMapping mapping = { channelName, {} };
+
+		// Position
+
+		auto positionIt = find( positionChannels.begin(), positionChannels.end(), channelName );
+		if( positionIt != positionChannels.end() && positionIt - positionChannels.begin() < 3 )
+		{
+			mapping.destinations.push_back( {
+				"P",
+				V3fVectorData::staticTypeId(),
+				static_cast<size_t>( positionIt - positionChannels.begin() )
+			} );
+			numPositionMappings++;
+		}
+
+		// Custom primitive variables
+
+		if( StringAlgo::matchMultiple( channelName, primitiveVariables ) )
+		{
+			const int colorIndex = ImageAlgo::colorIndex( channelName );
+			if( colorIndex >= 0 && colorIndex <= 2 )
+			{
+				// Map R, G and B to the components of colour primitive variables.
+				const string layerName = ImageAlgo::layerName( channelName );
+				mapping.destinations.push_back( {
+					layerName == "" ? "Cs" : layerName,
+					Color3fVectorData::staticTypeId(),
+					(size_t)colorIndex
+				} );
+			}
+			else
+			{
+				// Map everything else to individual float primitive variables.
+				mapping.destinations.push_back( {
+					channelName,
+					FloatVectorData::staticTypeId(),
+					0
+				} );
+			}
+		}
+
+		// Width
+
+		if( channelName == widthChannel )
+		{
+			mapping.destinations.push_back( {
+				"width",
+				FloatVectorData::staticTypeId(),
+				0
+			} );
+			haveWidthMapping = true;
+		}
+
+		// Alpha for skipping transparent pixels
+
+		if( channelName == "A" && ignoreTransparent )
+		{
+			mapping.destinations.push_back( {
+				"__imageToPointsAlpha__",
+				FloatVectorData::staticTypeId(),
+				0
+			} );
+			haveAlphaMapping = true;
+		}
+
+		if( mapping.destinations.size() )
+		{
+			result.push_back( mapping );
+		}
+	}
+
+	if( numPositionMappings < 3 && positionChannels.size() > numPositionMappings )
+	{
+		throw IECore::Exception( "Position channels are missing from the input image" );
+	}
+
+	if( !widthChannel.empty() and !haveWidthMapping )
+	{
+		throw IECore::Exception( "Width channels is missing from the input image" );
```

Comment:
"Width channel is missing from the input image"

Reply:
Also fixed in https://github.com/GafferHQ/gaffer/commit/a771ceb6acd146ad66ab1336f7969510a03a75c3.